cocotb/_gpi_embed.py

```diff
@@ -0,0 +1,41 @@
+import logging
+import os
+import importlib
+from functools import reduce
+
+
+def _filter_from_c(logger_name, level):
+    return logging.getLogger(logger_name).isEnabledFor(level)
+
+
+def _log_from_c(logger_name, level, filename, lineno, msg, function_name):
+    """
+    This is for use from the C world, and allows us to insert C stack
+    information.
+    """
+    logger = logging.getLogger(logger_name)
+    if logger.isEnabledFor(level):
+        record = logger.makeRecord(
+            logger.name,
+            level,
+            filename,
+            lineno,
+            msg,
+            None,
+            None,
+            function_name
+        )
+        logger.handle(record)
+
+
+def _load_entry():
+    """
+    Parses entry point specified by :envar:`COCOTB_ENTRY_POINT`, gathers entry point information
+    """
+    entry_point_str = os.environ.get("COCOTB_ENTRY", "cocotb:_initialise_testbench")
+    entry_module_str, entry_func_str = entry_point_str.split(":", 1)
+    entry_module = importlib.import_module(entry_module_str)
+    entry_func = reduce(getattr, entry_func_str.split('.'), entry_module)
+    entry_sim_event = getattr(entry_module, "_sim_event")
```

Comment:
I wonder whether we could lift the restriction on the name of this function - would it be reasonable to have the entry point call `simulator.get_sim_event_handler(...)`?

Reply:
No, it wouldn't be too late, `_sim_event` is only callable by the simulator after initialization, so if a callback is registered in the entry function, all is well.

Reply:
Well, there is an issue here, GPI expects to call `embed_sim_event` in the case of a sim event, which in turn calls the saved Python function `pEvent`. The simulator module does not have access to set anything in `gpi_embed`, it only works with the GPI interface.

We would have to change this interface so that GPI has a `register_sim_event_callback` function. `gpi_embed` would use the function to register `embed_sim_event` instead the function being at a known location.

Reply:
Continuing discussion in #1446.